tests/apps/courses/test_models_organization_role.py

```diff
@@ -0,0 +1,106 @@
+        {
+            "ADMIN": {
+                "django_permissions": [
+                    "courses.add_course",
+                    "cms.add_page",
+                    "cms.change_page",
+                ]
+            }
+        },
+    )
+    def test_models_page_role_django_permissions_group_admin(self, *_):
+        """
+        A group should be automatically created the first time the page role is saved.
+        It should get admin django permissions if the ADMIN role is defined.
+        """
+        page = PageFactory(title__title="My page")
+        role = PageRole(page=page, role="ADMIN")
+
+        with self.assertRaises(ObjectDoesNotExist):
+            self.assertIsNone(role.group)
+
+        # Now save the object and check that the group field was populated
+        role.save()
+        self.assertEqual(role.group.name, "Admin of My page")
+        self.assertEqual(role.group.permissions.count(), 3)
+        self.assertEqual(
+            [p.codename for p in role.group.permissions.all()],
+            ["add_page", "change_page", "add_course"],
+        )
+
+    @mock.patch.dict(
+        PAGE_ROLES, {"EDITOR": {"django_permissions": ["cms.change_page"]}}
+    )
+    def test_models_page_role_django_permissions_group_editor(self, *_):
+        """
+        A group should be automatically created the first time the page role is saved.
+        It should get editor django permissions if the EDITOR role is defined.
+        """
+        page = PageFactory(title__title="My page")
+        role = PageRole(page=page, role="EDITOR")
+
+        with self.assertRaises(ObjectDoesNotExist):
+            self.assertIsNone(role.group)
+
+        # Now save the object and check that the group field was populated
+        role.save()
+        self.assertEqual(role.group.name, "Editor of My page")
+        self.assertEqual(role.group.permissions.count(), 1)
+        self.assertEqual(
+            [p.codename for p in role.group.permissions.all()], ["change_page"]
+        )
+
+    @mock.patch.dict(
+        PAGE_ROLES, {"EDITOR": {"django_permissions": ["cms.change_page"]}}
+    )
+    def test_models_page_role_django_permissions_group_editor_extra(self, *_):
+        """
+        It should be possible to add or remove permissions to a group after its creation.
+        """
+        role = PageRoleFactory(role="EDITOR")
+
+        # Adding a permission
+        self.assertEqual(role.group.permissions.count(), 1)
+        extra_permission = PermissionFactory()
+        role.group.permissions.add(extra_permission)
+        self.assertEqual(role.group.permissions.count(), 2)
+
+        # Removing a permission
+        change_page = Permission.objects.get(codename="change_page")
+        role.group.permissions.remove(change_page)
+        self.assertEqual(role.group.permissions.count(), 1)
+        self.assertEqual(
+            [p.codename for p in role.group.permissions.all()],
+            [extra_permission.codename],
+        )
```

Comment:
You haven't tested that the `create_group` method is only called once on an instance. Or, if you consider my alternative, we need to check that we reset appropriate permissions once called.

Reply:
@jmaupetit I added some checks in the test, please take a look.